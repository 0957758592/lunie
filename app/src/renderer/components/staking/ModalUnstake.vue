<template lang="pug">
  .modal-stake#modal-stake
    //- Header
    .stake-header
      img.icon(class='stake-atom' src="~assets/images/cosmos-logo.png")
      span.tm-modal-title Unstake
      .tm-modal-icon.tm-modal-close(@click="close()")
        i.material-icons close

    //- To
    tm-form-group.stake-form-group(
      field-id='to' field-label='To')
      tm-field#to(
        readonly v-model="to")

    //- Amount
    tm-form-group.stake-form-group(
      field-id='amount'
      field-label='Amount'
    )
      tm-field#amount(
        :max="maximum"
        :min="0"
        type="number"
        v-focus
        v-model="amount")

    //- Footer
    .stake-footer
      tm-btn(
        @click.native="close"
        size="lg"
        value="Cancel"
      )
      tm-btn(
        @click.native="onUnstake"
        color="primary"
        size="lg"
        value="Unstake"
      )
</template>

<script>
import Modal from "common/TmModal"
import { TmBtn, TmField, TmFormGroup, TmFormMsg } from "@tendermint/ui"

export default {
  props: [`maximum`, `to`],
  components: {
    Modal,
    TmBtn,
    TmField,
    TmFormGroup,
    TmFormMsg
  },
  data: () => ({
    amount: 0
  }),
  methods: {
    close() {
      this.$emit("update:showModalUnstake", false)
    },
    onUnstake() {
      this.$emit(`submitUndelegation`, {
        amount: this.amount
      })

      this.close()
    }
  }
}
</script>

<style lang="stylus">
@import '~variables'

.modal-stake
  background var(--app-nav)
  display flex
  flex-direction column
  height 50%
  justify-content space-between
  left 50%
  padding 2em
  position fixed
  top 50%
  width 40%
  z-index z(modal)

  .stake-header
    align-items center
    display flex

    .stake-atom
      height 3em
      width 3em

  .stake-form-group
    display block
    padding 0

  .stake-footer
    display flex
    justify-content flex-end

    button
      margin-left 1em
</style>
