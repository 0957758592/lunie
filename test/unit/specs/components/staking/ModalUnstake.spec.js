"use strict"

import htmlBeautify from "html-beautify"
import { createLocalVue, mount } from "@vue/test-utils"
import Vuelidate from "vuelidate"
import ModalUnstake from "staking/ModalUnstake"

// Create an example stake modal window.
const Wrapper = () => {
  const localVue = createLocalVue()
  localVue.use(Vuelidate)

  return mount(ModalUnstake, {
    localVue,
    propsData: {
      maximum: 100,
      to: `cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctplpn3au`
    }
  })
}

test(`renders correctly`, () => {
  expect(htmlBeautify(Wrapper().html())).toMatchSnapshot()
})

test(`the "amount" field defaults to 0`, () => {
  expect(Wrapper().vm.amount).toEqual(0)
})

test(`display the 'To' address`, () => {
  expect(Wrapper().find(`#to`).element.value).toEqual(
    `cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctplpn3au`
  )
})

test(`unstake button emits the unstake signal`, () => {
  const wrapper = Wrapper()
  wrapper.setData({ amount: 50 })
  wrapper.vm.onUnstake()

  expect(wrapper.emittedByOrder()).toEqual([
    {
      name: `submitUndelegation`,
      args: [
        {
          amount: 50
        }
      ]
    },
    {
      name: `update:showModalUnstake`,
      args: [false]
    }
  ])
})
