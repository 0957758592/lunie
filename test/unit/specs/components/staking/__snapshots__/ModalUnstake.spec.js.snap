// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
"<div id=\\"modal-unstake\\" class=\\"modal-unstake\\">
   <div class=\\"stake-header\\">
      <img src=\\"~assets/images/cosmos-logo.png\\" class=\\"icon stake-atom\\"><span class=\\"tm-modal-title\\">Unstake</span>
      <div class=\\"tm-modal-icon tm-modal-close\\">
         <i class=\\"material-icons\\">close</i>
      </div>
</div>
<div class=\\"stake-form-group tm-form-group\\">
   <!---->
   <label for=\\"to\\" class=\\"tm-form-group__label\\">To</label>
   <div class=\\"tm-form-group__field\\">
      <input class=\\"tm-field\\" id=\\"to\\" readonly=\\"readonly\\">
   </div>
</div>
<div class=\\"stake-form-group tm-form-group\\">
   <!---->
   <label for=\\"amount\\" class=\\"tm-form-group__label\\">Amount</label>
   <div class=\\"tm-form-group__field\\">
      <input type=\\"number\\" max=\\"100\\" min=\\"0\\" class=\\"tm-field\\" id=\\"amount\\">
   </div>
</div>
<div class=\\"stake-footer\\">
   <button class=\\"tm-btn\\"><span class=\\"tm-btn__container tm-btn--size-lg tm-btn--primary\\">
   <!---->
   <!---->
   <span class=\\"tm-btn__value\\">Unstake</span></span></button>
</div>
</div>"
`;
